dex-it/src/test/scala/com/wavesplatform/it/sync/compat/DatabaseBackwardCompatTestSuite.scala

```diff
@@ -13,38 +14,77 @@ import scala.util.Random
 
 @DexMultipleVersions
 class DatabaseBackwardCompatTestSuite extends BackwardCompatSuiteBase {
+
+  val assetPairs = List(ethWavesPair, wavesUsdPair)
+  val additionalAssetPair = btcUsdnPair
+
   "Database backward compatibility test" in {
-    val assetPairs = List(ethWavesPair, wavesUsdPair)
+
     val twoAccountsOrdersGen = Gen.oneOf(
       orderGen(matcher, alice, assetPairs),
       orderGen(matcher, bob, assetPairs)
     )
 
+    markup("generate place commands for 100 limit and 100 market orders")
     // To ignore not important errors
     val placeCommands = Random.shuffle(
       Gen
         .containerOfN[Vector, Order](200, twoAccountsOrdersGen).sample
         .get
-        .map(MatcherCommand.Place(dex2, _))
+        .zipWithIndex
+        .map(p => MatcherCommand.Place(dex2, p._1, (p._2 % 2) == 0)) // make half of it market orders
     )
+
+    markup("place orders at DEX2")
     val orders = executePlaces(placeCommands)
     withClue("orders.size: ") {
-      orders.size shouldBe >(0)
+      orders.size shouldBe >(100) // if its > 100, then there will be not only limit or place orders
     }
 
     orders.groupBy(_.assetPair).valuesIterator.foreach { orders =>
       dex2.api.waitForOrder(orders.last)(_.status != Status.NotFound)
     }
+
+    markup("cancel one order at DEX2")
+    val orderToCancel = orders.head
```

Comment:
This probably doesn't work, because this order could be matched, thus we don't save the Cancel command to the queue and don't test here.

One approach is to get only active orders of alice and bob, pick a random active and cancel it just by `dex1.api.cancel`